tests/dist/python/run_dist_objects.py

```diff
@@ -180,6 +180,62 @@ def test_dist_embedding(g):
     dist_embedding_check_existing(num_nodes)
 
 
+##########################################
+############# DistOptimizer ##############
+##########################################
+
+
+def dist_optimizer_check_store(g, num_nodes):
+    rank = g.rank()
+    try:
+        emb = dgl.distributed.DistEmbedding(
+            num_nodes, 1, name="optimizer_test", init_func=zeros_init
+        )
+        emb2 = dgl.distributed.DistEmbedding(
+            num_nodes, 1, name="optimizer_test2", init_func=zeros_init
+        )
+        emb_optimizer = dgl.distributed.optim.SparseAdam([emb, emb2], lr=0.1)
+        if rank == 0:
+            name_to_state = {}
+            for _, emb_states in emb_optimizer._state.items():
+                for state in emb_states:
+                    name_to_state[state.name] = F.uniform(
+                        state.shape, F.float32, F.cpu(), 0, 1
+                    )
+                    state[
```

Comment:
This is test code so the size of graph is controlled and won't be too large. Check it distributively may increase complexity and less coverage of the code (because the code for split nodes should also be tested), so check it at one rank is more robust and clear.